src/lando/api/legacy/workers/uplift_worker.py

```diff
@@ -0,0 +1,235 @@
+                revision.commit_message,
+                revision.author,
+                revision.timestamp,
+            )
+
+        for uplift_revision in job.revisions.all():
+            self.handle_new_commit_failures(
+                apply_patch, repo, job, scm, uplift_revision
+            )
+            new_commit = scm.describe_commit()
+            logger.debug(f"Created new commit {new_commit}")
+
+        # On success: create patches.
+        response = self.moz_phab_uplift(
+            job, user.profile.phabricator_api_key, base_revision
+        )
+
+        # Retrieve created revision IDs and tip revision ID.
+        commits = response["commits"]
+        created_revision_ids = [int(commit["rev_id"]) for commit in commits]
+        tip_revision_id = created_revision_ids[-1]
+
+        _, created = UpliftRevision.objects.get_or_create(
+            revision_id=tip_revision_id,
+            defaults={"assessment": multi_request.assessment},
+        )
+        if not created:
+            UpliftRevision.objects.filter(revision_id=tip_revision_id).update(
+                assessment=multi_request.assessment
+            )
+
+        # Trigger a Celery task to update the form on Phabricator.
+        self.call_task(
+            set_uplift_request_form_on_revision,
+            tip_revision_id,
+            multi_request.assessment.to_conduit_json_str(),
+            user.id,
+        )
+
+        job.created_revision_ids = created_revision_ids
+        job.transition_status(JobAction.SUCCESS)
+
+        return created_revision_ids
+
+    def notify_success(
+        self,
+        repo_label: str,
+        job_url: str,
+        recipient_email: str,
+        created_revision_ids: list[int],
+    ) -> None:
+        if not recipient_email:
+            return
+        revision_labels = [f"D{rev_id}" for rev_id in created_revision_ids]
+        self.call_task(
+            send_uplift_success_email,
+            recipient_email,
+            repo_label,
+            job_url,
+            revision_labels,
+        )
+
+    def notify_uplift_failure(
+        self,
+        job: UpliftJob,
+        repo_label: str,
+        job_url: str,
+        recipient_email: str,
+        reason: str,
+    ) -> None:
+        if not recipient_email:
+            return
+
+        failure_reason = job.error or reason
+        self.call_task(
+            send_uplift_failure_email,
+            recipient_email,
+            repo_label,
+            job_url,
+            failure_reason,
+        )
+
+    def moz_phab_uplift(self, job: UpliftJob, api_key: str, base_revision: str) -> dict:
+        """Run `moz-phab uplift` on the repo."""
+        target_repo = job.target_repo
+
+        env = os.environ.copy()
+        env["MOZPHAB_PHABRICATOR_API_TOKEN"] = api_key
+
+        with tempfile.NamedTemporaryFile(
+            encoding="utf-8", mode="w+", suffix="json"
+        ) as f_output:
+            try:
+                subprocess.run(
```

Comment:
The moz-phab command should probably be abstracted into its own helper method that takes the `target_repo` and `base_revision` as inputs, and a success or error output that can be handled in this method. Alternatively, the outputs can be handled in `apply_and_uplift` instead, and the implementation of this method could just be slightly modified.